x-pack/legacy/plugins/lens/public/app_plugin/url_helper.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { DashboardConstants } from '../../../../../../src/legacy/core_plugins/kibana/public/dashboard/dashboard_constants';
+
+const EMPTY_DASHBOARD_PATTERN = /(.*#\/dashboard\?)(.*)/;
+const DASHBOARD_WITH_ID_PATTERN = /(.*#\/dashboard\/.*\?)(.*)/;
+
+/** *
+ * Returns base path from dashboard url
+ * eg.
+ * input: http://localhost:5601/lib/app/kibana#/dashboard?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(description:'',filters:!(),fullScreenMode:!f,options:(hidePanelTitles:!f,useMargins:!t),panels:!(),query:(language:kuery,query:''),timeRestore:!f,title:'',viewMode:edit)
+ * output: http://localhost:5601/lib/app/kibana#
+ * @param url
+ */
+export function getKibanaBasePathFromDashboardUrl(url: string | undefined): string | null {
+  if (!url) {
+    return null;
+  }
+  let regex = RegExp(/(.*)(\/dashboard\?)/).exec(url);
+  if (regex) {
+    return regex[1];
+  }
+  regex = RegExp(/(.*)(\/dashboard)/).exec(url);
+  if (regex) {
+    return regex[1];
+  }
+  return null;
+}
+
+/** *
+ * Returns dashboard URL with added embeddableType and embeddableId query params
+ * eg.
+ * input: url: http://localhost:5601/lib/app/kibana#/dashboard?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now)), embeddableId: 12345, embeddableType: 'lens'
+ * output: http://localhost:5601/lib/app/kibana#dashboard?embeddableType=lens&embeddableId=12345&_g=(refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))
+ * @param url dasbhoard absolute url
+ * @param embeddableId id of the saved visualization
+ * @param embeddableType 'lens' or 'visualization'
+ */
+export function addEmbeddableToDashboardUrl(
+  url: string | undefined,
+  embeddableId: string,
+  embeddableType: 'lens' | 'visualization'
```

Comment:
It seems this utility is purely within the `lens` code. Do we need to have `visualization` available here?

Reply:
No, not for now. But I was thinking of replacing the mash in visualization editor with this code
(and removing it from Lens namespace). But will remove this for now.